packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -449,6 +449,12 @@ export abstract class BaseSegment extends MergeNode implements ISegment {
 					0x044 /* "On annotate ack, missing segment property manager!" */,
 				);
 				this.propertyManager.ackPendingProperties(opArgs.op);
+				if (
+					!this.propertyManager.hasPendingProperties() &&
+					Object.entries(this.properties!).length === 0
```

Comment:
Use keys instead of entries here and in properties.ts and snapshotv1.ts and snapshotLegacy.ts?